azavea/pwd-river-dispatches
# Styled header/subtitles and "sensors"

## Overview
This PR addresses the need to style metrics, headers/subtitles, and the "sensors" page suggested in issue #24.

To do make these changes, I also added an `assets/` folder that contains `League Gothic Condensed` ([an open source typeface](https://www.theleagueofmoveabletype.com/league-gothic) that resembles the one used elsewhere in the application) and two illustration options for the Sensors page. 

## Screenshots
### Turbidity with reading from the Schuylkill
![turbidity_schuylkill](https://user-images.githubusercontent.com/5672295/27876633-cee03578-6185-11e7-9825-653d81e0756e.png)
### Turbidity with reading from the Delaware
![turbidity_bu_delaware](https://user-images.githubusercontent.com/5672295/27876645-d906b27a-6185-11e7-838e-ab1cbcc12e70.png)
### Turbidity with no sensor reading
![turbidity_bu_none](https://user-images.githubusercontent.com/5672295/27876675-e959c464-6185-11e7-8302-6615ea1817b4.png)
### Sensor (option 1)
![sensor_1](https://user-images.githubusercontent.com/5672295/27876684-f350e0f6-6185-11e7-82d7-7e6502d9835a.png)
### Sensor (option 2)
![sensor_2](https://user-images.githubusercontent.com/5672295/27876697-f9a453f2-6185-11e7-9418-b611e4909da8.png)

## Notes
- Working with Scott and Arianna, we shortened the text down for each screen.
- To style the Sensors page, I tried to use the `if(!metric)` statement within `window.onload` in order to give the page's `.container` a class name of "sensor". Please let me know if this was a bad way to do this! 
- I'm not sure if I need all of the fallbacks I included in my `@font-face` declaration. 
- If neither illustration fit what you imagined for Sensors, please let me know, @ajrobbins!

## Testing
- Get this branch, then `server`
- Load the spokes without making any changes and verify that you see data for the Schuylkill and appropriate headers and subscripts
- Drop `.usgs` from the `PRIMARY_SENSOR_URL` constant, then refresh the app. visit the spokes and verify that you see the values, headers, and subscripts from the Delaware River.
- Drop `.usgs` from the `BACKUP_SENSOR_URL` const, refresh the app, then visit the spokes again. Verify that you see the fallback values and text.
- Switch out the illustration being used by changing `background-image: url("assets/Sensor_Illustration_opt2.png");` to `background-image: url("assets/Sensor_Illustration_opt1.png");`
`
